packages/sidecar/src/utils/docker-compose-parser.ts

```diff
@@ -0,0 +1,55 @@
+import { existsSync, readFileSync } from "node:fs";
+import { parse as parseYaml } from "yaml";
+import { logger } from "../logger.js";
+
+const COMPOSE_FILE_NAMES = ["docker-compose.yaml", "docker-compose.yml", "compose.yml", "compose.yaml"] as const;
+
+/**
+ * Find the compose file in the current directory
+ */
+export function findComposeFile(): string | null {
+  for (const fileName of COMPOSE_FILE_NAMES) {
+    if (existsSync(fileName)) {
+      return fileName;
+    }
+  }
+  return null;
+}
+
+/**
+ * Find the override file for a given compose file
+ */
+export function findOverrideFile(composeFile: string): string | null {
+  const ext = composeFile.endsWith(".yaml") ? ".yaml" : ".yml";
+  const baseName = composeFile.replace(/\.(yaml|yml)$/, "");
```

Comment:
Not sure if this is vibe coded but we can make this part easier by changing `COMPOSE_FILE_NAMES` above a bit like this:

```ts
const COMPOSE_FILE_BASE_NAMES = ["compose", "docker-compose"];
const COMPOSE_FILE_EXTENSIONS = [".yml", ".yaml"];
```

Then we can just loop over both and build both file names (compose file and override file)